### PR TITLE
Tescase illustrating a segfault

### PR DESCRIPTION
**Pull request by aslakhellesoy**
Hi @brasmusson this PR contains a feature file that causes the `gherkin` CLI to segfault.

Thanks @cyocum for providing it! For context, @cyocum is toying with building a new Cucumber implementation in OCaml, taking advantage of the CLI parser.

The fix can go into this repo and I'll sync it back to the [monorepo](https://github.com/cucumber/cucumber).

### CODE REVIEW

**Comment by brasmusson** [comment]
Now that bug in the Pickles compiler is fixed.

File: src/compiler.c [hunk 1]
```diff
@@ -91,11 +91,11 @@ int Compiler_compile(Compiler* compiler, const GherkinDocument* gherkin_document
...
                     continue;
                 }
-                const PickleTags* tags = create_pickle_tags(feature->tags, scenario_outline->tags, example_table->tags);
                 int l;
                 for (l = 0; l < example_table->table_body->row_count; ++l) {
                     const TableRow* table_row = &example_table->table_body->table_rows[l];
                     const PickleLocations* locations = PickleLocations_new_double(table_row->location.line, table_row->location.column, scenario_outline->location.line, scenario_outline->location.column);
+                    const PickleTags* tags = create_pickle_tags(feature->tags, scenario_outline->tags, example_table->tags);
                     PickleSteps* steps = (PickleSteps*)malloc(sizeof(PickleSteps));
                     steps->step_count = scenario_outline->steps->step_count +  + background_step_count;
```
Comment (cyocum) [inline src/compiler.c:100]: I don't want to be out of order here but is this line correct? There are two additions here.  Should there only be one?

Comment (aslakhellesoy) [reply src/compiler.c:100]: From what I can tell @brasmusson moved line 94 to line 98 that's all. One deletion, one addition.

Comment (cyocum) [reply src/compiler.c:100]: @aslakhellesoy it's not that. It is line 100.  There are two pluses there where I think there should only be one:

```
steps->step_count = scenario_outline->steps->step_count +  + background_step_count;
```
-----------------------------------------------------------------------------------------^^^^^

Comment (aslakhellesoy) [reply src/compiler.c:100]: Oh that's weird I agree. It was there before this PR though.

Comment (cyocum) [reply src/compiler.c:100]: Indeed. How you would like me proceed? Should I file a bug?

Comment (aslakhellesoy) [reply src/compiler.c:100]: I'll remove a `+` and if it passes I'll just commit it :-P

**Pull request merged**